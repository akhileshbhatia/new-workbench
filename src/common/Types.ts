export type ChromeStorageQueryData = {
  data: string
};

export type QueryDataMap = Map<string, Map<number, ChromeStorageQueryData>>;
