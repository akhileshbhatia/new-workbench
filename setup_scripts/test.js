export const a = 'TEST EXPORT';