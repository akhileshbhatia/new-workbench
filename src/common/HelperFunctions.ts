export function getDataForPath(path: string) {
  return new Promise((resolve, reject) => {
    try {
      chrome.storage.local.get(path, (data) => {
        path === null ? resolve(data) : resolve(data[path]);
      });
    } catch (err) {
      reject(err);
    }
  });
}

export function clearStorage() {
  return new Promise((resolve, reject) => {
    try {
      chrome.storage.local.clear(() => resolve());
    } catch (err) {
      reject(err);
    }
  });
}

export function setDataToPath(path: string, data: string | Object) {
  return new Promise((resolve, reject) => {
    try {
      chrome.storage.local.set({ [path]: data }, () => {
        if (chrome.runtime.lastError) {
          reject('Error in setting data to chrome storage');
        }
        resolve();
      });
    } catch (err) {
      reject(err);
    }
  });
}

export function serializeMap(map) {
  return JSON.stringify(Array.from(map.entries()));
}

export function deserializeToMap(data: any) {
  return new Map(JSON.parse(data));
}

export function deserializeData(data: string) {
  const dateMap = deserializeToMap(data);
  const finalMap = new Map();
  for (const [date, info] of dateMap.entries()) {
    const newValue = deserializeToMap(info);
    finalMap.set(date, newValue);
  }
  return { output: finalMap };
}

export function getFormattedTime(timestamp: string): string {
  const date = new Date(+timestamp * 1000);
  return `${date.getHours()}:${date.getMinutes()}`;
}
