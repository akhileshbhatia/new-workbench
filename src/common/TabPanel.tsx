import React, { ReactNode } from 'react';

type TabPanelProps = {
  children: ReactNode
  value: number;
  index: number;
}

export function TabPanel(props: TabPanelProps): JSX.Element {
  const { value, index, children } = props;
  return (
    <div>
      {value === index && <div>{children}</div>}
    </div>
  );
}

