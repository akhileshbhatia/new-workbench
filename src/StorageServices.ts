import {
  getFormattedDateAndTimestamp,
  setDataToPath,
  serializeMap,
  getDataForPath
} from './common/HelperFunctions';
import { extensionStateKey } from './common/Constants';

export async function addToStorage(
  dataForPath: Map<string, Map<number, Record<string, unknown>>>,
  currentPathName: string,
  newData: Record<string, string>): Promise<void> {

  const { formattedDate, timestamp: newTimestamp } = getFormattedDateAndTimestamp();
  const dataForPathArray = [...dataForPath];
  /* 
    Since we can only insert at TODAY's date (and not in past or future), that means we only need to check 
    the latest date if it exists. If the date exists, that means we need to update that date's map. 
    If it doesn't exist, that means we need a new entry for that date
  */
  const latestDate = !!dataForPathArray.length && dataForPathArray[0][0];
  if (latestDate && (latestDate === formattedDate)) {
    // convert the existing details map to array to insert into beginning
    const existingMapArray = [...dataForPathArray[0][1]];
    existingMapArray.unshift([newTimestamp, newData]); // insert the new record
    dataForPathArray[0][1] = new Map(existingMapArray); // convert back to map
  } else {
    const newMap = new Map<number, Record<string, unknown>>();
    newMap.set(newTimestamp, newData);
    dataForPathArray.unshift([formattedDate, newMap]); // insert new record in the data
  }
  const dateMap = new Map<string, string>();
  for (const [date, info] of dataForPathArray) { // Serialize data again before storing
    dateMap.set(date, serializeMap<number, Record<string, unknown>>(info));
  }

  await setDataToPath(currentPathName, serializeMap<string, string>(dateMap));

}

export async function updateExtensionState(currentPathName: string, newState: boolean): Promise<void> {
  const currentExtensionStates = await getDataForPath(extensionStateKey) || {};
  currentExtensionStates[currentPathName] = newState;
  await setDataToPath(extensionStateKey, currentExtensionStates);
}
