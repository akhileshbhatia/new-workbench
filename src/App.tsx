import React, { ReactElement, useState, ChangeEvent } from 'react';
import './app.scss';
import {
  Drawer,
  IconButton,
  makeStyles,
  Theme,
  TextField,
  InputAdornment
} from '@material-ui/core';
import ChevronRight from '@material-ui/icons/ChevronRight';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import SearchIcon from '@material-ui/icons/Search';

import clsx from 'clsx';
import { updateExtensionState } from './StorageServices';
import { setDataToPath, serializeMap } from './common/HelperFunctions';
import QueryAccordion from './QueryAccordion';


const drawerWidth = 240;

const useStyles = makeStyles((theme: Theme) => ({
  root: {
    display: 'flex',
    width: '100%'
  },
  menuButton: {
    marginRight: theme.spacing(2)
  },
  hide: {
    display: 'none'
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0
  },
  drawerPaper: {
    width: drawerWidth
  },
  drawerHeader: {
    display: 'flex',
    alignItems: 'center',
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
    justifyContent: 'flex-end'
  },
  inputField: {
    margin: theme.spacing(1),
    width: '90%',
    height: '70px'
  }
}));

export default function App(props): ReactElement {
  const { output, defaultDrawerState, currentPathName } = props;
  const classes: Record<string, string> = useStyles();
  const [drawerOpen, setDrawerOpen] = useState(defaultDrawerState);
  const [searchTerm, setSearchTerm] = useState('');
  const [allData, setAllData] = useState(output);

  const updateDrawerState = async (newState: boolean) => {
    setDrawerOpen(newState);
    await updateExtensionState(currentPathName, newState);
  }

  const handleSearchTermUpdate = (event: ChangeEvent<HTMLInputElement>) => {
    setSearchTerm(event.target.value);
  }

  const handleDelete = async (timestamp: string, dateToDeleteFrom: string) => {
    const detailsMap = allData.get(dateToDeleteFrom);
    detailsMap.delete(timestamp);
    if (detailsMap.size > 0) {
      allData.set(dateToDeleteFrom, detailsMap); // Reset the new map for the same date
    } else {
      allData.delete(dateToDeleteFrom); // remove the date if it has no data
    }
    setAllData(() => new Map([...allData]));
    const dataToStore = new Map<string, string>();
    for (const [date, details] of allData.entries()) {
      dataToStore.set(date, serializeMap<number, Record<string, unknown>>(details));
    }
    await setDataToPath(currentPathName, serializeMap<string, string>(dataToStore));
  }

  return (
    <div className={classes.root}>
      <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={() => updateDrawerState(true)}
        edge="end"
        className={clsx(classes.menuButton, drawerOpen && classes.hide)}
      >
        <ChevronRight />
      </IconButton>
      <Drawer
        className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={drawerOpen}
        classes={{
          paper: classes.drawerPaper
        }}
      >
        <div className={classes.drawerHeader}>
          <IconButton onClick={() => updateDrawerState(false)}>
            <ChevronLeftIcon />
          </IconButton>
        </div>
        <div hidden={allData.size !== 0}>
          <h2>Please add queries/searches to see them here!</h2>
        </div>
        <div hidden={allData.size === 0}>
          <TextField
            variant="outlined"
            className={classes.inputField}
            value={searchTerm}
            onChange={handleSearchTermUpdate}
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">
                  <SearchIcon />
                </InputAdornment>
              )
            }}
          />
        </div>
        <div>
          {
            [...allData.keys()].map(date => {
              const entries = [...allData.get(date).entries()];
              const props = { date, entries, searchTerm, handleDelete };
              return <QueryAccordion key={date} {...props} />;
            })
          }
        </div>
      </Drawer>
    </div >
  )
}


