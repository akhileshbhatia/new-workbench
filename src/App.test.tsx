import React from 'react';
import { AppProps, App } from './App';
import {
  RenderResult,
  render,
  fireEvent,
  getByTestId,
  waitFor,
  getAllByTestId,
  queryByTestId
} from '@testing-library/react';
import { QueryDataMap, TimeDetailsMap } from './common/Types';
import { updateExtensionState } from './StorageServices';
import { setDataToChromeStorage, serializeMap } from './common/HelperFunctions';
import { getRandomTimestamp } from './test/HelperFunctions';

jest.mock('./StorageServices', () => {
  return {
    updateExtensionState: jest.fn()
  };
});

jest.mock('./common/HelperFunctions', () => {
  const { getHoursAndMinsFromTimestamp, serializeMap } = jest.requireActual('./common/HelperFunctions');
  return {
    setDataToChromeStorage: jest.fn(),
    getHoursAndMinsFromTimestamp,
    serializeMap
  };
});

describe('App', () => {
  let component: RenderResult;
  let queryDataMap: QueryDataMap;
  let timeDetailsMap: TimeDetailsMap;

  beforeEach(() => {
    jest.useFakeTimers();
    queryDataMap = new Map();
    timeDetailsMap = new Map();
  });

  const init = (dataArray: string[] = [], defaultDrawerState = true, currentPathName = 'query'): void => {
    dataArray.map(data => {
      timeDetailsMap.set(getRandomTimestamp(), { data });
      queryDataMap.set('12 March 2021', timeDetailsMap);
    });
    const props: AppProps = {
      output: queryDataMap,
      defaultDrawerState,
      currentPathName
    };
    component = render(<App {...props} />);
  };

  const getDrawerDivVisibility = (): string => {
    const drawer = component.container.getElementsByClassName('MuiDrawer-root')[0].firstElementChild;
    return window.getComputedStyle(drawer).visibility;
  };

  // Deletes the top most item it can find
  const deleteItem = (): void => {
    fireEvent.click(getAllByTestId(component.container, 'delete-icon')[0]);
  };

  it('should match snapshot', () => {
    init();
    expect(component).toMatchSnapshot();
  });

  it('should hide the data div when there is no data', () => {
    init();
    expect(getByTestId(component.container, 'no-data-msg').hasAttribute('hidden')).toBeFalsy();
    expect(getByTestId(component.container, 'data-div').hasAttribute('hidden')).toBeTruthy();
  });

  it('should open the drawer by default', () => {
    init();
    expect(getDrawerDivVisibility()).toBe('visible');
  });

  it('should close the drawer by default', () => {
    init(undefined, false);
    expect(getDrawerDivVisibility()).toBe('hidden');
  });

  it('should toggle the drawer when button is clicked', async () => {
    const testPathName = 'test-path';
    init(undefined, undefined, testPathName);
    fireEvent.click(getByTestId(component.container, 'close-drawer-button'));
    await waitFor(() => {
      expect(getDrawerDivVisibility()).toBe('hidden');
      expect(updateExtensionState).toHaveBeenCalledWith(testPathName, false);
    });
    fireEvent.click(getByTestId(component.container, 'open-drawer-button'));
    await waitFor(() => {
      expect(getDrawerDivVisibility()).toBe('visible');
      expect(updateExtensionState).toHaveBeenCalledWith(testPathName, true);
    });
  });

  it('should delete elements as expected', async () => {
    const testPathName = 'test-path';
    const dataArray = ['select name from info', 'select id, date from details', 'select test from app_test'];
    init(dataArray, true, testPathName);
    // Check that all the item are present
    expect(getAllByTestId(component.container, 'accordion-details').length).toBe(dataArray.length);
    // Delete first item
    deleteItem();
    // Prepare expected output
    const dataToStore = new Map<string, string>();
    for (const [date, details] of queryDataMap.entries()) {
      dataToStore.set(date, serializeMap<number, Record<string, unknown>>(details));
    }
    await waitFor(() => {
      expect(setDataToChromeStorage).toHaveBeenCalledTimes(1);
      expect(setDataToChromeStorage).toHaveBeenCalledWith(testPathName, serializeMap<string, string>(dataToStore));
      // UI should now have one less item
      expect(getAllByTestId(component.container, 'accordion-details').length).toBe(dataArray.length - 1);
    });

    // Delete both the remaining items
    deleteItem();
    deleteItem();

    await waitFor(() => {
      expect(setDataToChromeStorage).toHaveBeenCalledTimes(3);
      expect(setDataToChromeStorage).toHaveBeenLastCalledWith(testPathName, serializeMap<string, string>(new Map()));
      expect(queryByTestId(component.container, 'accordion-details')).toBeNull();
    });
  });
});
