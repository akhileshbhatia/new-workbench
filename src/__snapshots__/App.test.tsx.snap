// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <button
          aria-label="open drawer"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 makeStyles-hide-3 MuiIconButton-colorInherit MuiIconButton-edgeEnd"
          data-testid="open-drawer-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
          >
            <div
              class="makeStyles-drawerHeader-6"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                data-testid="close-drawer-button"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              data-testid="no-data-msg"
            >
              <h2>
                Please add queries/searches to see them here!
              </h2>
            </div>
            <div
              data-testid="data-div"
              hidden=""
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-inputField-7"
                data-testid="search-text-field"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-9"
                      style="width: 0.01px;"
                    >
                      <span>
                        â€‹
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <header
              class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation4"
            >
              <div
                class="MuiTabs-root"
              >
                <div
                  class="MuiTabs-scroller MuiTabs-fixed"
                  style="overflow: hidden;"
                >
                  <div
                    class="MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        All
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiTab-wrapper"
                      >
                        Bookmarked
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="PrivateTabIndicator-root-12 PrivateTabIndicator-colorPrimary-13 MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
            </header>
            <div>
              <div
                class="MuiBox-root MuiBox-root-16"
              >
                <div />
              </div>
            </div>
            <div
              hidden=""
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <button
        aria-label="open drawer"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 makeStyles-hide-3 MuiIconButton-colorInherit MuiIconButton-edgeEnd"
        data-testid="open-drawer-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-4"
      >
        <div
          class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-5 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        >
          <div
            class="makeStyles-drawerHeader-6"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              data-testid="close-drawer-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            data-testid="no-data-msg"
          >
            <h2>
              Please add queries/searches to see them here!
            </h2>
          </div>
          <div
            data-testid="data-div"
            hidden=""
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-inputField-7"
              data-testid="search-text-field"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedStart MuiOutlinedInput-adornedStart"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiOutlinedInput-inputAdornedStart"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-8 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-9"
                    style="width: 0.01px;"
                  >
                    <span>
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation4"
          >
            <div
              class="MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed"
                style="overflow: hidden;"
              >
                <div
                  class="MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected MuiTab-fullWidth"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper"
                    >
                      All
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiTab-wrapper"
                    >
                      Bookmarked
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="PrivateTabIndicator-root-12 PrivateTabIndicator-colorPrimary-13 MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </header>
          <div>
            <div
              class="MuiBox-root MuiBox-root-16"
            >
              <div />
            </div>
          </div>
          <div
            hidden=""
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
